aws/table_aws_iam_access_advisor.go

```diff
@@ -94,6 +96,18 @@ func listAccessAdvisor(ctx context.Context, d *plugin.QueryData, _ *plugin.Hydra
 	granularity := "ACTION_LEVEL"
 	principalArn := d.KeyColumnQuals["principal_arn"].GetStringValue()
 
+	getCommonColumnsCached := plugin.HydrateFunc(getCommonColumns).WithCache()
+	commonData, err := getCommonColumnsCached(ctx, d, h)
+	if err != nil {
+		return nil, err
+	}
+	commonColumnData := commonData.(*awsCommonColumnData)
+
+	// check if principalArn is empty or if the account id in the principalArn is not same with the account
+	if principalArn == "" || len(strings.Split(principalArn, ":")) < 4 || strings.Split(principalArn, ":")[4] != commonColumnData.AccountId {
```

Comment:
I don't think we want to return nil if the arn has no account id, since that would mean we could no longer evaluate aws-managed policies, such as `arn:aws:iam::aws:policy/AmazonS3FullAccess`?